### ISSUE TITLE
Roslyn binaries not copied properly on publish in 1.0.6

### ISSUE THREAD

**Issue by onyxmaster** [aspnet/RoslynCodeDomProvider#13]
Using web publish from VS 2015 fails with the following:
```
C:\Windows\Microsoft.NET\Framework\v4.0.30319\aspnet_compiler.exe -v / -p <snip>\obj\Release\AspnetCompileMerge\Source -c -d <snip>\obj\Release\AspnetCompileMerge\TempBuildDir 
ASPNETCOMPILER(0,0): Error ASPRUNTIME: Could not find a part of the path '<snip>\obj\Release\AspnetCompileMerge\Source\bin\roslyn\csc.exe'.
```
Rolling back to 1.0.5 works.
I believe that the fix for #2 broke the web publish scenario at least with VS 2015.

This is a local path web publish with the following settings:
```xml
<PackageAsSingleFile>false</PackageAsSingleFile>
<PrecompileBeforePublish>true</PrecompileBeforePublish>
<ExcludeGeneratedDebugSymbol>false</ExcludeGeneratedDebugSymbol>
<UseMerge>true</UseMerge>
<WDPMergeOption>MergeAllOutputsToASingleAssembly</WDPMergeOption>
<EnableUpdateable>false</EnableUpdateable>
```

**Comment by ThomasPrioul** [comment]
I have experienced problems too with 1.0.6 on a full .NET app using Roslyn compiler for scripting purposes.
The roslyn binaries get copied to "D:\bin\" instead of "D:\Path\To\Project\Output\bin\roslyn"...
Took me a while to figure out where they were copied to.

**Comment by Jinhuafei** [comment]
Thanks for reporting the issue. I'll take a look.

**Comment by stephenpatten** [comment]
@Jinhuafei , Is this resolved? Looks like you referenced another ticket that refers back to this one.. Thank you

**Comment by Jinhuafei** [comment]
Sent [PR 14](https://github.com/aspnet/RoslynCodeDomProvider/pull/14)

**Comment by KarlZ** [comment]
@onyxmaster You're picture made my day... To me it looks like an expression of, "Come'on, REALLY?" (Which is certainly how I feel now...)

**Comment by nicholasyin** [comment]
Had the same issue. Downgrading to 1.0.5 works. thanks guys.

**Comment by chadik79** [comment]
Having the same problem, downgrading to 1.0.5 works

**Comment by robk410** [comment]
whatever this issue is, it has totally hosed our build. Even after rolling back, I still get an error about not being able to find 1.0.6 assembly. Changed build to clean source and directories and it still happens. Out of options here. Fix ASAP.

**Comment by vdevappa** [comment]
I have seen issues where on the build server, for some reason unless you delete the downloaded source code, it keeps old files. Delete manually and then try to rebuild again.

**Comment by robk410** [comment]
Yeah, it was the web.config that was the hold up. Roll back to 1.0.5 does work for me now. VS2k15u3

**Comment by ricardag** [comment]
Happens here with **VS2017 15.2 (26430.16) Release** when publishing to file, and "**Precompile during publishing**" is checked. If we uncheck this option, the publish succeeds.

Not on all publishing, just some profiles. And deleting/recreating profile doesn't helps.

**Comment by TheDestinyGroup** [comment]
Hi,

I had started a question on this last week over on Stack Overflow.

It can be found at:

https://stackoverflow.com/questions/45538776/v1-0-6-of-microsoft-codedom-providers-dotnetcompilerplatform/

Rohit submitted an answer to my question with a work-around.

The issue appears to lie with the Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props file in the build\net45 directory.

Hopefully the developers can work out why the changes they've made in this release related to how the RoslyCompilerFiles are copied don't appear to be working in VS2015 or VS2017.

Les

**Comment by Jinhuafei** [comment]
@TheDestinyGroup If you can't wait for the new release, you can copy the CS in [this PR](https://github.com/aspnet/RoslynCodeDomProvider/pull/14) to workaround the issue for now.

**Comment by sherif-elmetainy** [comment]
Had same problem. Downgrading to 1.0.5 worked.

**Comment by ScottRFrost** [comment]
Same issue here.  .csproj created in VS2017 that had never touched VS2015.  Like others, uninstalling and then installing 1.0.5 worked.

1.0.6 should probably be pulled from nuget.

**Comment by IvanAlekseev** [comment]
We have the same issue with 1.0.7.
1.0.5 works fine

**Comment by Jinhuafei** [comment]
@IvanAlekseev can you give us more details? We tested 1.0.7 version with all the publishing scenarios in VS and didn't see any issue.

**Comment by erincerol** [comment]
@Jinhuafei We have used via VSTS App Service Publish and Azure App Service GitHub deployment. Both have the same issue.

**Comment by Jinhuafei** [comment]
@erincerol thanks. I'll take a look.

**Comment by RyanCEI** [comment]
FWIW I had exactly the same problem using VS Azure Publish.  I uninstalled 1.0.5 directly then installed 1.0.7 clean back into the project and it fixed the problem.  

I don't think it's the publish method but rather the copying of the compiler into the publish directory that is failing.  Runs fine locally even if the pub is broken.  I have diff's of my csproj, packages.config and web.config that I can DM you if you like.  I can also roll back and reproduce the problem exactly if you have an update you'd like to test.

**Comment by antisniperspy** [comment]
1.0.7 Fixed the issue for me in team services. Thanks

**Comment by Jinhuafei** [comment]
I opened a new issue to track the publish issues that 1.0.7 release doesn't fix.

**Comment by titobf** [comment]
1.0.7 worked for me when publishing to azure using web deploy

**Comment by knopa** [comment]
had same issue with teamcity and octopus on 1.0.7

**Comment by Jinhuafei** [comment]
@knopa do you mind sharing the build log with me?

**Comment by slewis74** [comment]
@Jinhuafei here's a simple web app project that illustrates what I think @knopa is seeing
[RoslynWebApplication1.zip](https://github.com/aspnet/RoslynCodeDomProvider/files/1240313/RoslynWebApplication1.zip)

To repro:
- open the solution and restore the nuget packages
- run build.cmd from the command line

OctoPack isn't packing any of the Roslyn files because they aren't there (I'm seeing an empty Roslyn folder under bin)

I had a look at `Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props` difference between 1.0.5 and 1.0.7 and this looked like a possible suspect

```<Target Name="IncludeRoslynCompilerFilesToFilesForPackagingFromProject" BeforeTargets="PrepareForRun" >```

in 1.0.5 it used to read

```<Target Name="IncludeRoslynCompilerFilesToItemGroup" AfterTargets="ResolveAssemblyReferences" >```

If I'm understanding correctly, it's only going to include the Roslyn files if it's preparing to run, which the build presumably isn't doing when executed from command line, TC etc?

**Comment by knopa** [comment]
@Jinhuafei 
Here is build log with 1.0.7 version - https://www.dropbox.com/s/xp1dduva4ca867k/demo_build.zip?dl=0

**Comment by Jinhuafei** [comment]
@slewis74  Thanks. I'm able to repro the issue with your app. The problem is due to copying Roslyn bits after OctoPacking. I'll look into a fix.

**Comment by Jinhuafei** [comment]
@knopa looks like the issue is same as the one @slewis74 mentioned above.

**Comment by mobius127** [comment]
Hi everyone, with version 1.0.7.0 still fails when I try to publish in Azure...

**Comment by Jinhuafei** [comment]
@mobius127 Can you share the msbuild log(please set the verbosity to diag)?

**Comment by Airn5475** [comment]
I rolled back from v1.0.7 to v1.0.6 and it worked fine.
I also tried uninstalling it altogether and fresh installing v1.0.7 and that worked as well.

**Comment by mobius127** [comment]
@Jinhuafei Here

Build started 30/08/2017 14:43:19.
Overriding target "GetFrameworkPaths" in project "C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets" with target "GetFrameworkPaths" from project "C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.NetFramework.CurrentVersion.targets".
Overriding target "SatelliteDllsProjectOutputGroup" in project "C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets" with target "SatelliteDllsProjectOutputGroup" from project "C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319\Microsoft.WinFx.targets".
The target "MvcBuildViews" listed in a BeforeTargets attribute at "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\Web\Microsoft.Web.Publishing.targets (852,131)" does not exist in the project, and will be ignored.
Project "D:\Users\fcgonzal\Documents\Visual Studio 2015\Projects\TestToken\WebApiToken\WebApiToken.csproj" on node 1 (go target(s)).
Building with tools version "14.0".
Project file contains ToolsVersion="4.0". This toolset may be unknown or missing, in which case you may be able to resolve this by installing the appropriate version of MSBuild, or the build may have been forced to a particular ToolsVersion for policy reasons. Treating the project as if it had ToolsVersion="14.0". For more information, please see http://go.microsoft.com/fwlink/?LinkId=293424.
Target "ValidateMSBuildToolsVersion" skipped, due to false condition; ('$(BuildingProject)' == 'true') was evaluated as ('false' == 'true').
Target "_CheckForInvalidConfigurationAndPlatform" in file "C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets" from project "D:\Users\fcgonzal\Documents\Visual Studio 2015\Projects\TestToken\WebApiToken\WebApiToken.csproj" (entry point):
Set Property: _InvalidConfigurationMessageText=The OutputPath property is not set for project 'WebApiToken.csproj'.  Please check to make sure that you have specified a valid combination of Configuration and Platform for this project.  Configuration='Debug'  Platform='AnyCPU'.
Set Property: _InvalidConfigurationMessageText=The OutputPath property is not set for project 'WebApiToken.csproj'.  Please check to make sure that you have specified a valid combination of Configuration and Platform for this project.  Configuration='Debug'  Platform='AnyCPU'.  You may be seeing this message because you are trying to build a project without a solution file, and have specified a non-default Configuration or Platform that doesn't exist for this project.
Task "Error" skipped, due to false condition; ( '$(_InvalidConfigurationError)' == 'true' ) was evaluated as ( '' == 'true' ).
Task "Warning" skipped, due to false condition; ( '$(_InvalidConfigurationWarning)' == 'true' ) was evaluated as ( '' == 'true' ).
Using "Message" task from assembly "Microsoft.Build.Tasks.Core, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
Task "Message"
  Task Parameter:Text=Configuration=Debug
  Task Parameter:Importance=Low
  Configuration=Debug
Done executing task "Message".
Task "Message"
  Task Parameter:Text=Platform=AnyCPU
  Task Parameter:Importance=Low
  Platform=AnyCPU
Done executing task "Message".
Task "Error" skipped, due to false condition; ('$(OutDir)' != '' and !HasTrailingSlash('$(OutDir)')) was evaluated as ('bin\' != '' and !HasTrailingSlash('bin\')).
Task "Error" skipped, due to false condition; ('$(BaseIntermediateOutputPath)' != '' and !HasTrailingSlash('$(BaseIntermediateOutputPath)')) was evaluated as ('obj\' != '' and !HasTrailingSlash('obj\')).
Task "Error" skipped, due to false condition; ('$(IntermediateOutputPath)' != '' and !HasTrailingSlash('$(IntermediateOutputPath)')) was evaluated as ('obj\Debug\' != '' and !HasTrailingSlash('obj\Debug\')).
Done building target "_CheckForInvalidConfigurationAndPlatform" in project "WebApiToken.csproj".
D:\Users\fcgonzal\Documents\Visual Studio 2015\Projects\TestToken\WebApiToken\WebApiToken.csproj : error MSB4057: The target "go" does not exist in the project.
Done Building Project "D:\Users\fcgonzal\Documents\Visual Studio 2015\Projects\TestToken\WebApiToken\WebApiToken.csproj" (go target(s)) -- FAILED.

Deferred Messages

Detailed Build Summary
======================
    
  
  ============================== Build Hierarchy (IDs represent configurations) =====================================================
  Id                  : Exclusive Time   Total Time   Path (Targets)
  -----------------------------------------------------------------------------------------------------------------------------------
  0                   : 0.347s           0.347s       D:\Users\fcgonzal\Documents\Visual Studio 2015\Projects\TestToken\WebApiToken\WebApiToken.csproj (go) 
  
  ============================== Node Utilization (IDs represent configurations) ====================================================
  Timestamp:            1        Duration   Cumulative
  -----------------------------------------------------------------------------------------------------------------------------------
  636396937991484595:   0        0,346s     0,346s ######
  -----------------------------------------------------------------------------------------------------------------------------------
  Utilization:          100,0    Average Utilization: 100,0

Build FAILED.

"D:\Users\fcgonzal\Documents\Visual Studio 2015\Projects\TestToken\WebApiToken\WebApiToken.csproj" (go target) (1) ->
  D:\Users\fcgonzal\Documents\Visual Studio 2015\Projects\TestToken\WebApiToken\WebApiToken.csproj : error MSB4057: The target "go" does not exist in the project.

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.38

**Comment by Jinhuafei** [comment]
@mobius127 from the error message "The target "go" does not exist in the project.", it doesn't seem related to the DotNetCompilerPlatform nupkg.

**Comment by mobius127** [comment]
@Jinhuafei but when I'm trying to publish in Azure my web application, fails because of it

**Comment by Jinhuafei** [comment]
@mobius127 if you uninstall DotNetCompilerPlatform, are you able to publish your app to Azure?

**Comment by BobbyCannon** [comment]
Uninstalling 1.0.7 then reinstall 1.0.7 will publish the assemblies but only once. The next publish will not publish them. This is using a local file path publish.

**Comment by BobbyCannon** [comment]
There's two ways to get it to deploy. Seems switching publish profiles forces it to redeploy, however if you do a publish profile the "deletes all files" and deploy twice the roslyn folder only is there on first deploy. It's not there any other deploys after. If you clean the site it will also redeploy without having to switch publish profiles. Rebuild will not work but clean will.

**Comment by BobbyCannon** [comment]
My guess is many people may not being seeing this issue if they do not publish with "delete existing files" flag turned on. A temporary work around is to just uncheck "delete existing files" when working locally.

**Comment by Jinhuafei** [comment]
@BobbyCannon thanks for the repro. I was able to repro the issue by publishing twice. The second publish doesn't build the project(nothing changed in the project), hence the copy rolsyn files task is not executed. Will look into a fix.

**Comment by ttrevorr** [comment]
Can we re-open this issue? Or create a new issue to track this since it clearly isnt fixed?

BTW I have the same issues with 1.0.7 with Teamcity and Octopus.

**Comment by Jinhuafei** [comment]
@ttrevorr we are tracking the publish related issue through issue #18

**Comment by akash3456** [comment]
Is there a workaround on this issue besides copying and pasting the roslyn folder into the bin, i have tried using the version 1.0.5 of the dom compiler and have also set csc.exe as environment variable, however i checked through MSBuild im using c#5 i thought this could be the answer.

**Comment by Jinhuafei** [comment]
@akash3456 would you please elaborate your issue? Are you talking about compiling the code behind file? Do you install RoslynCodeDomProvider nupkg in your project?

**Comment by akash3456** [comment]
So when I try to build a web project into an msi I was expecting to see the Roslyn folder in the bin folder of the project of installed files. But this was missing after installing code dom provider nuget package this didn't fix the issue but a manual intervention was required to build the Roslyn files from the source and add them in the msi.

**Comment by mac2000** [comment]
Can confirm also, at moment fighting with TeamCity and OctoPack, reverting package to 1.0.5 gives me roslyn folder inside bin, 1.0.7 - roslyn is missing.

**Comment by brgrz** [comment]
Seems fixed with 1.0.8 because with 1.0.7 and our TFSs build the /bin/roslyn folder was always missing after builds, now it is there.

**Comment by willbush** [comment]
Seems to me every version after 1.0.5 does not copy roslyn into the bin folder even on a new simple console application. It work for me in 1.0.5, but it has the bug that was fixed in 1.0.6 with the hard coded path.

**Comment by willbush** [comment]
Turns out if your project file does not define the variable "WebProjectOutputDir" it will fail to copy Roslyn over and it continues on error.

See the issue I opened: https://github.com/aspnet/RoslynCodeDomProvider/issues/21

**Comment by RickStrahl** [comment]
Running into this issue in non-Web projects as well. WPF app in this case, same problem with 1.08. Reverting to 1.05 works.

**Comment by willbush** [comment]
@RickStrahl see my previous comment and referenced issue. Adding "WebProjectOutputDir" variable to our csproj file worked in our WPF app.

**Comment by akash3456** [comment]
Ok thanks for identifying this as an issue. Will it work if I do specify webprojectoutputdir in settings?

**Comment by willbush** [comment]
@akash3456 Yes it will work, but it's a bit hacky.

**Comment by akash3456** [comment]
That’s not bad to be fair thanks for the effort

**Comment by Rzpeg** [comment]
Upgrading to 1.0.8 hasn't solved the issue. We are using the "PackageReference" Nuget format

**Comment by timmkrause** [comment]
Having the same issue on TeamCity. If I run the publish profile locally `roslyn` is missing. The proejct uses the "classic" `.csproj` (non-SDK) format and `PackageReference`s with `ToolsVersion="15.0"`.

What I have removed when upgrading to `PackageReference`s was

```
<Import Project="..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.1.0.8\build\net45\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props" Condition="Exists('..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.1.0.8\build\net45\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props')" /> 
<Import Project="..\packages\Microsoft.Net.Compilers.2.6.1\build\Microsoft.Net.Compilers.props" Condition="Exists('..\packages\Microsoft.Net.Compilers.2.6.1\build\Microsoft.Net.Compilers.props')" /> 
```
because from my understanding it is not required anymore and with `PackageReference`s the path to `..\packages` is not correct anymore anyway.

Any new insights on how to solve this? Why is TeamCity behaving differently...

**Comment by Jinhuafei** [comment]
@Rzpeg  @timmkrause can you provide msbuild log?

**Comment by timmkrause** [comment]
@Jinhuafei I am sorry but unfortunately I am not allowed to do so...

What came to my mind while reading the log was that the last build step which builds the project with `/p:DeployOnBuild=true` and a publish profile (executes MSDeploy) does the following before copying other files:

```
[12:26:02] :			 [CopyAllFilesToSingleFolderForMsdeploy] Delete
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\csc.exe".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\csc.exe.config".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\csc.rsp".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\csi.exe".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\csi.exe.config".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\csi.rsp".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.Build.Tasks.CodeAnalysis.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.CodeAnalysis.CSharp.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.CodeAnalysis.CSharp.Scripting.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.CodeAnalysis.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.CodeAnalysis.Scripting.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.CodeAnalysis.VisualBasic.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.CSharp.Core.targets".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.DiaSymReader.Native.amd64.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.DiaSymReader.Native.x86.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.VisualBasic.Core.targets".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\Microsoft.Win32.Primitives.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.AppContext.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Collections.Immutable.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Console.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Diagnostics.FileVersionInfo.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Diagnostics.Process.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Diagnostics.StackTrace.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.IO.Compression.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.IO.FileSystem.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.IO.FileSystem.DriveInfo.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.IO.FileSystem.Primitives.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.IO.Pipes.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Reflection.Metadata.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Security.AccessControl.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Security.Claims.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Security.Cryptography.Algorithms.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Security.Cryptography.Encoding.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Security.Cryptography.Primitives.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Security.Cryptography.X509Certificates.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Security.Principal.Windows.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Text.Encoding.CodePages.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Threading.Thread.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.ValueTuple.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Xml.ReaderWriter.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Xml.XmlDocument.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Xml.XPath.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\System.Xml.XPath.XDocument.dll".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\vbc.exe".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\vbc.exe.config".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\vbc.rsp".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\VBCSCompiler.exe".
[12:26:02] :				 [Delete] Deleting file "obj\Integration\Package\PackageTmp\bin\roslyn\VBCSCompiler.exe.config".
```

I fixed it now via a `{projectName}.wpp.targets` with the following content which explicitly copies the Roslyn files and deploys them:

```
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Target Name="CustomCollectFiles" BeforeTargets="CopyAllFilesToSingleFolderForPackage;CopyAllFilesToSingleFolderForMsdeploy">
    <ItemGroup>
      <RoslynFiles Include="bin/roslyn/*.*"/>
      <FilesForPackagingFromProject Include="%(RoslynFiles.Identity);">
        <DestinationRelativePath>%(RelativeDir)%(Filename)%(Extension)</DestinationRelativePath>
      </FilesForPackagingFromProject>
    </ItemGroup>
  </Target>
</Project>
```

I know this is a bit hacky but I need to make progress. Will keep an eye on this issue for a better solution.

**Comment by dmarlow** [comment]
Ran into this with `/p:DeployOnBuild=true` and a publish profile. Not too keen on using the hack (how is that even referenced anyways?)
